### ISSUE TITLE
Gene-length cutoffs for codon-specific densities

### ISSUE THREAD

**Issue by shahpr** [riboviz/riboviz#254]
In the current setup, all genes with <200 codons are ignored from analysis of codon-specific ribosome-densities. This initial cutoff was decided based on large CHX-driven ramps in early yeast ribo-seq data. Since most new datasets wouldn't have these ramps, and we don't see them in mammalian datasets, this cutoff seems arbitrary and will likely bias our estimates.

To address this, we will need to do two things - 
1. Remove this filter in the `CalculateCodonSpecificRibosomeDensity` function.
2. Change the code in `script_for_transcript_annotation.Rmd` to include data for first 200 codons.

Finally, I would like to rewrite this code and make use of tidy functions as opposed to the current list based operations.

**Comment by ewallace** [comment]
Yes please! Finishing up #194 will make this easier; that replaces the .RData-stored list of codon positions with a tidy data frame, I was planning to attack later this week.

**Comment by shahpr** [comment]
Great. In that case, I will attack #194 first.

**Comment by ewallace** [comment]
Noting that @shahpr fixed:
> 2. Change the code in `script_for_transcript_annotation.Rmd` to include data for first 200 codons.

as part of pull request #287.

That leaves:
> 1. Remove this filter in the CalculateCodonSpecificRibosomeDensity function.
> rewrite this code and make use of tidy functions

Premal, how close are you to doing that? It makes my head hurt just to look at the list-based code.
